sdk/iot/provisioning/src/az_iot_provisioning_client.c

```diff
@@ -10,3 +10,175 @@
 
 #include <_az_cfg.h>
 
+static const az_span provisioning_service_api_version
+    = AZ_SPAN_LITERAL_FROM_STR("/api-version=" AZ_IOT_PROVISIONING_SERVICE_VERSION);
+
+static const az_span str_dps = AZ_SPAN_LITERAL_FROM_STR("$dps");
+static const az_span str_registrations = AZ_SPAN_LITERAL_FROM_STR("/registrations/");
+static const az_span str_put_iotdps_register
+    = AZ_SPAN_LITERAL_FROM_STR("PUT/iotdps-register/?$rid=1");
+static const az_span str_get_iotdps_get_operationstatus
+    = AZ_SPAN_LITERAL_FROM_STR("GET/iotdps-get-operationstatus/?$rid=1&operationId=");
+static const az_span str_res = AZ_SPAN_LITERAL_FROM_STR("res/");
+
+AZ_NODISCARD az_iot_provisioning_client_options az_iot_provisioning_client_options_default()
+{
+  return (az_iot_provisioning_client_options){ .user_agent = AZ_SPAN_NULL };
+}
+
+AZ_NODISCARD az_result az_iot_provisioning_client_init(
+    az_iot_provisioning_client* client,
+    az_span global_device_endpoint,
+    az_span id_scope,
+    az_span registration_id,
+    az_iot_provisioning_client_options const* options)
+{
+  AZ_PRECONDITION_NOT_NULL(client);
+  AZ_PRECONDITION_VALID_SPAN(global_device_endpoint, 1, false);
+  AZ_PRECONDITION_VALID_SPAN(id_scope, 1, false);
+  AZ_PRECONDITION_VALID_SPAN(registration_id, 1, false);
+
+  client->_internal.global_device_endpoint = global_device_endpoint;
+  client->_internal.id_scope = id_scope;
+  client->_internal.registration_id = registration_id;
+
+  if (options != NULL)
+  {
+    client->_internal.options.user_agent = options->user_agent;
+  }
+  else
+  {
+    client->_internal.options = az_iot_provisioning_client_options_default();
+  }
+
+  return AZ_OK;
+}
+
+// <id_scope>/registrations/<registration_id>/api-version=<service_version>
```

Comment:
>//  [](start = 0, length = 3)

Is there any guidline to put comments?

I prefer atleast the function comments to be 
/* */

Reply:
This was formatted according to the clang rules and it's showing an example of what this function will currently return.
This is _not_ a contract or function documentation - that's done in the declaration site.